### ISSUE TITLE
Since last update to pixplot package, pip install pixplot does not work properly

### ISSUE THREAD

**Issue by shenglih** [YaleDHLab/pix-plot#96]
on Mac it returned this installation error:
         TSNE<SplitTree, &(euclidean_distance(DataPoint const&, DataPoint const&))>::run(dou
ble*, int, int, double*, int, double, double, int, int, int, int, bool, int, double, double
, double*) in tsne.cpp.o
        TSNE<SplitTree, &(euclidean_distance_squared(DataPoint const&, DataPoint const&))>:
:run(double*, int, int, double*, int, double, double, int, int, int, int, bool, int, double
, double, double*) in tsne.cpp.o
        VpTree<DataPoint, &(euclidean_distance(DataPoint const&, DataPoint const&))>::build
FromPoints(int, int) in tsne.cpp.o
        VpTree<DataPoint, &(euclidean_distance_squared(DataPoint const&, DataPoint const&))
>::buildFromPoints(int, int) in tsne.cpp.o
    "_srand", referenced from:
        TSNE<SplitTree, &(euclidean_distance(DataPoint const&, DataPoint const&))>::run(dou
ble*, int, int, double*, int, double, double, int, int, int, int, bool, int, double, double
, double*) in tsne.cpp.o
        TSNE<SplitTree, &(euclidean_distance_squared(DataPoint const&, DataPoint const&))>:
:run(double*, int, int, double*, int, double, double, int, int, int, int, bool, int, double
, double, double*) in tsne.cpp.o
    "_time", referenced from:
        TSNE<SplitTree, &(euclidean_distance(DataPoint const&, DataPoint const&))>::run(dou
ble*, int, int, double*, int, double, double, int, int, int, int, bool, int, double, double
, double*) in tsne.cpp.o
        TSNE<SplitTree, &(euclidean_distance_squared(DataPoint const&, DataPoint const&))>:
:run(double*, int, int, double*, int, double, double, int, int, int, int, bool, int, double
, double, double*) in tsne.cpp.o
  ld: symbol(s) not found for architecture x86_64
  clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)
  make[2]: *** [/private/var/folders/sf/4y0f6rb52gg54rskg_jdn5zc0000gr/T/pip-install-lrmh_jmm/MulticoreTSNE/build/lib.macosx-10.14-x86_64-3.7/MulticoreTSNE/libtsne_multicore.so] Error 1
  make[1]: *** [CMakeFiles/tsne_multicore.dir/all] Error 2
  make: *** [all] Error 2
  
  ERROR: Cannot find make? See above errors.
  ----------------------------------------
  ERROR: Failed building wheel for MulticoreTSNE
  Running setup.py clean for MulticoreTSNE
Failed to build MulticoreTSNE
ERROR: Could not build wheels for hdbscan which use PEP 517 and cannot be installed directly

on linux 16.4 it returned 
Traceback (most recent call last):
  File "/usr/local/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
ImportError: No module named 'pip._internal.cli.main'

**Comment by duhaime** [comment]
@shenglih sorry about that! This is pretty interesting.

I think the Linux case is easier to diagnose. It seems your Linux pip has somehow broken. Could you please try to reinstall pip with `sudo easy_install pip`? Then try another install with `pip install pixplot`...

The Mac case is more mysterious. On that OS you might want to uninstall Anaconda if it's already installed, then install Anaconda afresh (that worked for [this user](https://github.com/DmitryUlyanov/Multicore-TSNE/issues/67)).

If you try one or both of those suggestions please feel free to let me know how they go and I'll be happy to follow up!

**Comment by shenglih** [comment]
@duhaime thanks so much! I've tried both now the error is with pandas...
ERROR: Could not find a version that satisfies the requirement pandas>=0.25.3 (from pointgrid>=0.0.2->pixplot) (from versions: 0.1, 0.2b0, 0.2b1, 0.2, 0.3.0b0, 0.3.0b2, 0.3.0, 0.4.0, 0.4.1, 0.4.2, 0.4.3, 0.5.0, 0.6.0, 0.6.1, 0.7.0rc1, 0.7.0, 0.7.1, 0.7.2, 0.7.3, 0.8.0rc1, 0.8.0rc2, 0.8.0, 0.8.1, 0.9.0, 0.9.1, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.13.0, 0.13.1, 0.14.0, 0.14.1, 0.15.0, 0.15.1, 0.15.2, 0.16.0, 0.16.1, 0.16.2, 0.17.0, 0.17.1, 0.18.0, 0.18.1, 0.19.0rc1, 0.19.0, 0.19.1, 0.19.2, 0.20.0rc1, 0.20.0, 0.20.1, 0.20.2, 0.20.3, 0.21.0rc1, 0.21.0, 0.21.1, 0.22.0, 0.23.0rc2, 0.23.0, 0.23.1, 0.23.2, 0.23.3, 0.23.4, 0.24.0rc1, 0.24.0, 0.24.1, 0.24.2)
ERROR: No matching distribution found for pandas>=0.25.3 (from pointgrid>=0.0.2->pixplot)

**Comment by duhaime** [comment]
@shenglih aha! I have a hunch you're running Python 2.7--is that right? It seems pandas stopped supporting Python 2.7 in version `0.24.2` (Python 2.7 was in fact deprecated just last month).

Are you able to switch to Python 3.5 or 3.6? Either should make development easier than 2.7 will...

I'd install Anaconda (on Linux and Windows), then create a Python 3.5 environment with:

```bash
conda create --name 3.5 python=3.5
```

That creates an environment named "3.5" that uses Python version 3.5. Then to use that environment, just run `source activate 3.5` (or, in more recent versions of Anaconda, I believe it's now `conda activate 3.5`). If you then try the installation, you should be good to go. 

Feel free to try that out and report back on how it goes!

**Comment by duhaime** [comment]
@shenglih I wanted to check in to see if you're still having troubles with pip. If so please feel free to follow up! If not, is it safe to close this issue?

**Comment by duhaime** [comment]
@shenglih I'm going to close this out but if you have any further trouble please feel free to follow up and we can help!

**Comment by shawngraham** [comment]
hi, 

i'm trying this on a windows 10 machine, conda installed. made new 3.7 environment, installed pixplot as per readme, getting

```
 Could not build wheels for hdbscan which use PEP 517 and cannot be installed directly
```
...?

**Comment by tianyilt** [comment]
> hi,
> 
> i'm trying this on a windows 10 machine, conda installed. made new 3.7 environment, installed pixplot as per readme, getting
> 
> ```
>  Could not build wheels for hdbscan which use PEP 517 and cannot be installed directly
> ```
> 
> ...?

I'm facing the same error. Have you solved this problem?

**Comment by kongxdesign** [comment]
> > hi,
> > i'm trying this on a windows 10 machine, conda installed. made new 3.7 environment, installed pixplot as per readme, getting
> > ```
> >  Could not build wheels for hdbscan which use PEP 517 and cannot be installed directly
> > ```
> > 
> > 
> > ...?
> 
> I'm facing the same error. Have you solved this problem?

I'm facing the same error. Have you solved this problem?

**Comment by shawngraham** [comment]
I have not. Will try to explore it more once i have a bit more time

**Comment by duhaime** [comment]
@shawngraham @kongxdesign sorry to have been out of the loop on this issue! It looks like one path to victory here might be to [install anaconda](https://www.anaconda.com/products/individual#Downloads) and then create a virtual environment:

```bash
conda create --name=3.7 python=3.7
source activate 3.7
```

Then, before trying to pip install pixplot, could you try to install hdbscan with conda using the following?

```bash
conda install -c conda-forge hdbscan
```

After you run that line, you _should_ be able to:

```
pip uninstall pixplot
pip install https://github.com/yaledhlab/pix-plot/archive/master.zip
```

If you get a chance to try that out, could I please ask you to let us know how it goes?

**Comment by Giuliagiorgi** [comment]
Hi, I'm working on WIndows 10. I've tried it and solved creating a virual environment in py 3.6 not 3.7

conda create --name=3.6 python=3.6
source activate 3.6
